### PR TITLE
Adding second pass median subtraction for grism spectra

### PR DESCRIPTION
**Pull request by pascaloesch**
Added functionality to do a second pass median subtraction after masking pixels with significant flux (e.g. emission lines) in the first pass. This ameliorates self-subtraction of lines.
Code could potentially be improved by using a convolution rather than box filtering for binned S/N estimates.

### CODE REVIEW

File: grizli/pipeline/auto_script.py [hunk 1]
```diff
@@ -2982,7 +2982,8 @@ def grism_prep(field_root='j142724+334246', PREP_PATH='../Prep', EXTRACT_PATH='.
             # and don't do any contamination modeling
             refine_niter = 0
             grp.subtract_median_filter(filter_size=median_filter_size, 
-                                       filter_central=median_filter_central)
+                                       filter_central=median_filter_central,
+                                       second_pass_filtering=second_pass_filtering, box_filter_sn=box_filter_sn, box_filter_width=box_filter_width)
```
Comment (gbrammer) [inline grizli/pipeline/auto_script.py:2986]: Can you put line breaks for each new parameter to match the general codestyle?

File: grizli/multifit.py [hunk 2]
```diff
@@ -1213,6 +1213,33 @@ def subtract_median_filter(self, filter_size=71, filter_central=10, revert=True,
                                                footprint=filter_footprint)
                 
                 filter_sci[~np.isfinite(filter_sci)] = 0
+
+            if second_pass_filtering==True:
```
Comment (gbrammer) [inline grizli/multifit.py:1217]: `if second_pass_filtering:`

**Comment by gbrammer** [comment]
Looks great, thanks!  I put a few tiny code style requests in the comments, and then it's probably good to go.

**Comment by pascaloesch** [comment]
Great! The latest commit should include these updates.

**Pull request merged**